src/app.ts

```diff
@@ -13,14 +17,102 @@ interface TemperatureSummary {
   average: number
 }
 
+class dataByTime implements TemperatureReading {
+  public dataTemperature: Array<number> = []
+  // I created a new Class by error not definition
+  public TSummary: ClassTSummary = new ClassTSummary(0)
+  constructor(
+    public time: Date,
+    public city: string,
+    public temperature: number,
+  ) {
+    this.dataTemperature.push(temperature)
+  }
+
+  addTmpe(num: number): void {
+    this.dataTemperature.push(num)
+  }
+}
+
+class ClassTSummary implements TemperatureSummary {
+  constructor(temperature: number) {
+    this.first = this.last = this.high = this.low = this.average = temperature
+  }
+  first
+  last
+  high
+  low
+  average
+}
+
+function existValue(data: Date) {
+  for (const d of allData) {
+    if (d.time.getTime() === data.getTime()) return true
+  }
+
+  return false
+}
+
+//resolving y updating sumary
+function getSummary(tmpSummary: ClassTSummary, value: number): ClassTSummary {
+  if (tmpSummary.first === -1) tmpSummary.first = value
+  if (tmpSummary.high < value) tmpSummary.high = value
+
+  if (tmpSummary.low > value) tmpSummary.low = value
+
+  tmpSummary.last = value
+  tmpSummary.average += value
+  return tmpSummary
+}
+
 export function processReadings(readings: TemperatureReading[]): void {
-  // add here your code
+  for (let i = 0; i < readings.length; i++) {
+    if (!existValue(readings[i].time)) {
+      // if reading[i] do not exist in allData
+
+      const tmpe: TemperatureReading = readings[i]
```

Comment:
You shouldn't need this type assignment here. If your typescript is correct it will infer this.